markdowns/21_distill_a_reader.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: tutorial
+colab: https://colab.research.google.com/github/deepset-ai/haystack-tutorials/blob/main/tutorials/21_distill_a_reader.ipynb
+toc: True
+title: "Distill a Reader"
+last_updated: 2022-11-30
+level: "intermediate"
+weight: 115
+description: Transfer a Reader's question answering ability to a smaller, more efficient model.
+category: "QA"
+aliases: ['/tutorials/distill-reader']
+download: "/downloads/21_distill_a_reader.ipynb"
+---
+    
+
+
+- **Level**: Advanced
+- **Time to complete**: 30 minutes
+- **Nodes Used**: `FARMReader`
+- **Goal**: Distil the question answering capabilities of the larger BERT base Reader model into a smaller TinyBERT Reader model.
+
+
+## Overview
+
+Model distillation is the process of teaching a smaller model to imitate the performance of a larger, better trained model. By distilling one model into another, you end up with a more computationally efficient version of the original with only a slight trade-off in accuracy. In this tutorial, you will learn how to perform one form of model distillation on Reader models in Haystack. Model distillation is a complex topic and an active area of research so if want to learn more about it, see [Model Distillation](https://docs.haystack.deepset.ai/docs/model_distillation).
+
+## Preparing the Colab Environment
+
+<details>
+- [Enable GPU Runtime in GPU](https://docs.haystack.deepset.ai/docs/enable-gpu-runtime-in-colab)
+- [Check if GPU is Enabled](https://docs.haystack.deepset.ai/docs/check-if-gpu-is-enabled)
+- [Set logging level to INFO](https://docs.haystack.deepset.ai/docs/set-the-logging-level)
+</details>
+
+
+## Installing Haystack
+
+To start, let's install the latest release of Haystack with `pip`:
+
+
+```bash
+%%bash
+
+pip install --upgrade pip
+pip install farm-haystack[colab]
+```
+
+## Augmenting Training Data
+
+Having more human annotated training data is useful at all levels of model training. However, intermediate layer distillation can benefit even from synthetically generated data, since it is a less exact type of training. In this tutorial, we'll be using the [`augment_squad.py` script](https://github.com/deepset-ai/haystack/blob/main/haystack/utils/augment_squad.py) to augment our dataset. It creates artificial copies of question answering samples by replacing randomly chosen words with words of similar meaning. This meaning similarity is determined by their vector representations in a GLoVe word embedding model.
+
+1. Download the `augment_squad.py` script.
+
+
+```python
+!wget https://raw.githubusercontent.com/deepset-ai/haystack/main/haystack/utils/augment_squad.py
+```
+
+2. Download a small slice of the SQuAD question answering database.
+
+
+```python
+from haystack.utils import fetch_archive_from_http
+
+doc_dir = "data/distil_a_reader"
+squad_dir = doc_dir + "/squad"
+
+s3_url = "https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/documents/squad_small.json.zip"
```

Comment:
This dataset can be moved from tutorial 2 to 21 for telemetry @julian-risch ?

Reply:
Yes can be down. To make this process less error-prone maybe we can have a list with old tutorials dataset URLs and what they become? This is not the only one that needs to change and it would be great if we could do it in one larger batch if posssible.